frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx

```diff
@@ -126,18 +133,24 @@ const BundleDetail = ({
             let stateUpdate = {
                 fileContents: null,
             };
+
             ['stdout', 'stderr'].forEach(function(name) {
                 if (info.contents.some((entry) => entry.name === name)) {
                     fetchRequests.push(
-                        fetchFileSummary(uuid, '/' + name).then(function(blob) {
-                            stateUpdate[name] = blob;
-                        }),
+                        fetchFileSummary(uuid, '/' + name)
+                            .then(function(blob) {
+                                stateUpdate[name] = blob;
+                            })
+                            .catch((e) => {
+                                setFetchingFileSummary(false);

```

Comment:
This isn't correct -- what if one request (fetching stdout) fails but then the second request (fetching stderr) is still in progress -- then `fetchingFileSummary` will be set to False (when it should still be True).

I'd suggest instead setting `fetchingFileSummary` to `0` at the beginning so it stores a counter that stores the number of pending requests. Let's replace the following:
- `setFetchingFileSummary(true)` -> `setFetchingFileSummary(f => f + 1)`
- `setFetchingFileSummary(false)` -> `setFetchingFileSummary(f => f - 1)`
- `if (!info || fetchingFileSummary)` -> `if (!info || fetchingFileSummary > 0)`

Finally, maybe rename `fetchingFileSummary` to `pendingFileSummaryFetches`?

Reply:
Thanks I have updated the state variable!